### ISSUE TITLE
[Broker] Remove limit on the number of triggers per broker  

### ISSUE THREAD

**Issue by lionelvillard** [knative-extensions/eventing-kafka-broker#337]
**Problem**
The broker controller seems to create a ConfigMap containing all trigger configurations, putting a hard limit on how many triggers can be handled by a single broker. 

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?

**Exit Criteria**
A measurable (binary) test that would indicate that the problem has been resolved.

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?

**Additional context (optional)**
Add any other context about the feature request here.

This is more a question than a feature request. Feel free to close if I misunderstood how it works.

**Comment by pierDipi** [comment]
Yes, I've thought about this for a while now.

/assign

cc @slinkydeveloper

**Comment by slinkydeveloper** [comment]
I do wonder, @pierDipi and @lionelvillard, do we have an estimate on how large that number is?

**Comment by pierDipi** [comment]
No that I'm aware of, I can find an upper bound in extreme situations.

**Comment by pierDipi** [comment]
N = 1200, the contract in _json_ is almost 1MB
```
n 1200 json 987166 proto 793789
```
N = 1500, the contract in _proto_ is almost 1MB
```
n 1500 json 1233766 proto 992089
```

https://gist.github.com/pierDipi/f2d6775c68d4c6b44b6fbfb6e73050e3

**Comment by slinkydeveloper** [comment]
@lionelvillard what kind of "numbers" your kafka source operates with usually?

**Comment by pierDipi** [comment]
Fixed by #3558
